src/mlpack/methods/kde/kde_rules_impl.hpp

```diff
@@ -130,7 +133,11 @@ Score(const size_t queryIndex, TreeType& referenceNode)
   const double minKernel = kernel.Evaluate(maxDistance);
   const double bound = maxKernel - minKernel;
 
-  if (bound <= (absError + relError * minKernel) / referenceSet.n_cols)
+  // Error tolerance of the current query point and reference node.
+  const double errorTolerance =
+      (absError + relError * minKernel) / referenceSet.n_cols;
+
+  if (bound <= accumError(queryIndex) + errorTolerance)
```

Comment:
It may be worth adding a comment for clarity here; something like

```
// We relax the bound for pruning by accumError(queryIndex), so that if there
// is any "leftover" error from the rest of the traversal, we can use it in this check.
```

or maybe something more elaborate, I am not sure, but I suspect someone reading this code will wonder what is going on here. :)